favicons (build with https://realfavicongenerator.net)

KristjanESPERANTO opened the pull request:
With the Favicon checker [1] you can check "more than a dozen settings to make sure it fits all major platforms". The result of openvegemap.netlib.re shows that there are some gaps [2]. This pull request should fill those gaps.

[1] https://realfavicongenerator.net/favicon_checker
[2] https://realfavicongenerator.net/favicon_checker?protocol=https&site=openvegemap.netlib.re%2F

Rudloff commented on img/browserconfig.xml:
```diff
@@ -0,0 +1,9 @@
+    <msapplication>
+        <tile>
+            <square150x150logo src="img/mstile-150x150.png"/>
+            <TileColor>#00aba9</TileColor>
```
Where does this teal color come from? It is not a color used in our interface.
In fact, the interface does not have many color so I think we should not force a color and let the browser use whatever native color it wants.

KristjanESPERANTO replied:
I couldn't find a more suitable color - feel free to choose another one. I don't have a good talent for colors.

Isn't it better if we determine a suitable color ourselves than let the system choose a random one?

Rudloff replied:
Couldn't we use the green from the logo (`#119352`)?

KristjanESPERANTO replied:
Yes, this should look good :slightly_smiling_face: I changed it in the PR.

Rudloff commented on index.html:
```diff
@@ -1,25 +1,36 @@
 <!DOCTYPE HTML>
 <html lang="en">
     <head>
-        <meta charset="UTF-8">
```
I see you made unrelated changes to the markup here, but it makes it harder to see what you added.

KristjanESPERANTO replied:
Yes, that was bad practice. I moved the charset tag back to the original position.

Rudloff commented on index.html:
```diff
@@ -1,25 +1,36 @@
+        <link rel="canonical" href="https://openvegemap.netlib.re/">
+        <link rel="preconnect" href="https://overpass-api.de/">
+
+        <!-- favicons (build with https://realfavicongenerator.net) -->
```
Can you please remove this?

KristjanESPERANTO replied:
Sure! I assume you mean the comment. Lines 17 and 18 were already there before. I removed this and an other comment.

Rudloff commented on index.html:
```diff
@@ -1,25 +1,36 @@
+        <link rel="apple-touch-icon" sizes="180x180" href="img/apple-touch-icon.png">
+        <link rel="icon" type="image/png" sizes="32x32" href="img/favicon-32x32.png">
+        <link rel="icon" type="image/png" sizes="16x16" href="img/favicon-16x16.png">
+        <link rel="mask-icon" href="img/safari-pinned-tab.svg" color="#00aba9">
```
If I understand correctly, this will make Safari use a teal circle as favicon?

KristjanESPERANTO replied:
That wasn't the intention. I've replaced the svg file. Now it should look like this in the preview of the generator:

![Ekrankopio de 2020-10-20 09-06-27](https://user-images.githubusercontent.com/35647502/96552649-571f1480-12b4-11eb-94c9-7eb2a8ba6a43.png)

Rudloff commented on js/poi.js:
```diff
@@ -164,6 +164,8 @@ function POI(tags) {
             return '🍸';
         case 'pub':
             return '🍺';
+        case 'ice_cream':
+            return '🍨';
```
Can you please move this to a separate PR (in order to make it easier to review)?

KristjanESPERANTO replied:
Done.

Rudloff replied:
Thanks, feel free to submit another PR for it!

Rudloff reviewed (changes requested):
Thanks for the PR!

However, I have some questions:
* Why does `apple-touch-icon.png` have non-transparent background?
* `favicon-32x32.png` is somehow larger and more pixelated than the original `logo_32.png` file.
* Shouldn't every PNG image be added to the manifest?

KristjanESPERANTO commented:
> * Why does `apple-touch-icon.png` have non-transparent background?

The generator explains it that way: _"Your favicon is transparent. This is usually a good thing, but not for iOS Web Clips which are supposed to be opaque." + "iOS will fill the transparent regions with black."_

Black isn't a suitable background color I think.

---

> * `favicon-32x32.png` is somehow larger and more pixelated than the original `logo_32.png` file.

I didn't notice any deterioration. See my screenshot here:

![Ekrankopio de 2020-10-20 09-27-49](https://user-images.githubusercontent.com/35647502/96554336-b3833380-12b6-11eb-99e2-7ccf192b69ce.png)

Regardless of my impression, I've just renamed the old `logo_32.png` and replaced `favicon-32x32.png` with this one.

---

> * Shouldn't every PNG image be added to the manifest?

I'm not sure, but I don't think this would be necessary.

---

I will try to queue up smaller pull requests in the future so that you have it easier to review.

Rudloff commented:
> Black isn't a suitable background color I think.

Fair enough. Maybe use a white background instead?

KristjanESPERANTO commented:
> Fair enough. Maybe use a white background instead?

White doesn't look good. Now I changed it also to the green from the logo. I think, that's a good solution :slightly_smiling_face:
